src/Npgsql/NpgsqlTypes/NpgsqlDate.cs

```diff
@@ -1,452 +1,235 @@
-
-        internal int DaysSinceEra => _daysSinceEra;
-
-        public bool IsLeapYear => IsLeap(Year);
-
-        public bool IsInfinity => _type == InternalType.Infinity;
-        public bool IsNegativeInfinity => _type == InternalType.NegativeInfinity;
-
-        public bool IsFinite
-            => _type switch {
-                InternalType.Finite           => true,
-                InternalType.Infinity         => false,
-                InternalType.NegativeInfinity => false,
-                _ => throw new InvalidOperationException($"Internal Npgsql bug: unexpected value {_type} of enum {nameof(NpgsqlDate)}.{nameof(InternalType)}. Please file a bug.")
-            };
-
-        #endregion
-
-        #region Internals
-
-        static int DaysForYears(int years)
-        {
-            //Number of years after 1CE (0 for 1CE, -1 for 1BCE, 1 for 2CE).
-            var calcYear = years < 1 ? years : years - 1;
-
-            return calcYear / 400 * DaysIn4Centuries //Blocks of 400 years with their leap and common years
-                   + calcYear % 400 / 100 * DaysInCentury //Remaining blocks of 100 years with their leap and common years
-                   + calcYear % 100 / 4 * DaysIn4Years //Remaining blocks of 4 years with their leap and common years
-                   + calcYear % 4 * DaysInYear //Remaining years, all common
-                   + (calcYear < 0 ? -1 : 0); //And 1BCE is leap.
-        }
-
-        static bool IsLeap(int year)
-        {
-            //Every 4 years is a leap year
-            //Except every 100 years isn't a leap year.
-            //Except every 400 years is.
-            if (year < 1)
-            {
-                year = year + 1;
-            }
-            return (year%4 == 0) && ((year%100 != 0) || (year%400 == 0));
-        }
-
-        #endregion
-
-        #region Arithmetic
+        /// <param name="year">The year (4713 BC through 5874897 AD).</param>
+        /// <param name="month">The month (1 through 12).</param>
+        /// <param name="day">The day (1 through the number of days in <paramref name="month"/>).</param>
+        public NpgsqlDate(int year, int month, int day) => _days = NpgsqlDateTime.ToDays(year, month, day);
 
-        public NpgsqlDate AddDays(int days)
-            => _type switch
-        {
-            InternalType.Infinity         => Infinity,
-            InternalType.NegativeInfinity => NegativeInfinity,
-            InternalType.Finite           => new NpgsqlDate(_daysSinceEra + days),
-            _ => throw new InvalidOperationException($"Internal Npgsql bug: unexpected value {_type} of enum {nameof(NpgsqlDate)}.{nameof(InternalType)}. Please file a bug.")
-        };
+        /// <summary>Represents the largest possible value of <see cref="NpgsqlDate"/>.</summary>
+        /// <value>The largest possible value of <see cref="NpgsqlDate"/>.</value>
+        public static NpgsqlDate MaxValue => new NpgsqlDate(NpgsqlDateTime.DaysMaxValue);
```

Comment:
nit - can use target-typed new (here and everywhere else).

```suggestion
        public static NpgsqlDate MaxValue => new(NpgsqlDateTime.DaysMaxValue);
```